src/components/KCodeBlock/CodeBlockIconButton.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <KTooltip
+    class="code-block-action-tooltip"
+    placement="bottomStart"
+    :popover-timeout="300"
+    :text="copyTooltipText"
+  >
+    <KButton
+      appearance="tertiary"
+      class="code-block-action-button"
+      :class="[`theme-${theme}`, { 'code-block-action-button-active': active }]"
+      v-bind="attrs"
+      @click="onClick"
+    >
+      <!-- TODO: [beta] update once icon slot is deprecated -->
+      <template #icon>
+        <slot />
+      </template>
+    </KButton>
+  </KTooltip>
+</template>
+
+<script setup lang="ts">
+import { ref, type PropType, useAttrs } from 'vue'
+import type { Theme } from '@/types'
+import { watch } from 'vue'
+import KButton from '@/components/KButton/KButton.vue'
+import KTooltip from '@/components/KTooltip/KTooltip.vue'
+
+const props = defineProps({
+  theme: {
+    type: String as PropType<Theme>,
+    default: 'light',
+  },
+  active: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+   * Only pass tooltip text if it's the copy button
+   */
+  copyTooltip: {
+    type: String,
+    default: '',
+  },
+})
+
+const copyTooltipText = ref<string>(props.copyTooltip)
+
+const attrs = useAttrs()
+
+const onClick = () => {
+  // if copy text is passed means it's the copy button - therefore change tooltip text to 'Copied!' on click
+  if (props.copyTooltip) {
+    copyTooltipText.value = 'Copied!'
+  }
+}
+
+watch(copyTooltipText, () => {
+  // change tooltip back to original after 3 seconds
+  setTimeout(() => {
+    copyTooltipText.value = props.copyTooltip
+  }, 3000)
+})
+</script>
+
+<style lang="scss" scoped>
+.code-block-action-tooltip {
+  // display: inline-block;
+}
+
+// expand upon original tertiary button styles changing some colors and states
+.code-block-action-button {
```

Comment:
**Issue**: As far as I can tell, the styles for this button now only apply to exactly this button. However, the code block has a `secondary-actions` slot which we use to (sometimes) slot in extra buttons. The styles here should also apply to those buttons. Otherwise, you end up needing to customize those styles in the consuming application to avoid this:

![image](https://github.com/Kong/kongponents/assets/5774638/af15b1d9-481a-4f77-8706-6cf3b4dd6318)

Reply:
For reference, how we use the `secondary-actions` slot:

https://github.com/kumahq/kuma-gui/blob/master/src/app/common/code-block/ResourceCodeBlock.vue#L19-L40

Reply:
Added export and docs for `KCodeBlockIconButton` component: https://deploy-preview-2100--kongponents.netlify.app/components/codeblock.html#kcodeblockiconbutton

Reply:
Nice, thanks, I’ll have a look now.

Reply:
Yep, I think I can adapt our implementation to use this helper component:
![image](https://github.com/Kong/kongponents/assets/5774638/4c176ef1-6512-4b41-8ad2-1dc7bc7d2a8b)

Reply:
Awesome! Notice that you don't need to add spacing to that component (margin) - the KCodeBlock slot styles take care of that for you.